sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DecimalPrecision.scala

```diff
@@ -42,8 +43,10 @@ import org.apache.spark.sql.types._
  *   e1 / e2      p1 - s1 + s2 + max(6, s1 + p2 + 1)      max(6, s1 + p2 + 1)
  *   e1 % e2      min(p1-s1, p2-s2) + max(s1, s2)         max(s1, s2)
  *   e1 union e2  max(s1, s2) + max(p1-s1, p2-s2)         max(s1, s2)
- *   sum(e1)      p1 + 10                                 s1
- *   avg(e1)      p1 + 4                                  s1 + 4
+ *
+ * When `spark.sql.decimalOperations.allowTruncat` is set to true, if the precision / scale needed
+ * are out of the range of available values, the scale is reduced up to 6, in order to prevent the
```

Comment:
it's better to push some reference about where we get this rule.

Reply:
if scale needs to be less than 6, we would fail the analysis, right?

Reply:
no, we would return the a type with scale 6 and in `CheckOverflow` all the numbers which don't fit will be translated to `NULL`, since this is current Spark behavior in such cases.

May I kindly ask you to elaborate a but more what you mean by "push some reference"? Thanks.

Reply:
sorry, typo, "put some reference"...

We can put a link to a document of a mainstream RDBMS and say this rule follows xxx...

Reply:
I did, but @gatorsmile told me to remove it: https://github.com/apache/spark/pull/20023#discussion_r159117817.

Reply:
Did any open source RDBMS have this rule?

Reply:
Actually we already referred a commercial RDBMS in L33...

Reply:
Hive has it, but in the documentation it is not explained. And in the comments it just references the blog I referenced, but then removed according to @gatorsmile's comment.